clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -6822,53 +7014,11 @@ class FreeFunctionPrinter {
       }
 
       const TemplateSpecializationType *TSTAsNonAlias =
-          TST->getAsNonAliasTemplateSpecializationType();
+          TST->getAsTemplateSpecializationTypeWithoutAliases(Context);
```

Comment:
So I was kind of expecting that the visitor soluiton would help us to avoid unwrapping and modifying types manually like `getAsTemplateSpecializationTypeWithoutAliases` does. Is that not the case? Could you please elaborate why do we need such a complex solution?

Reply:
This patch was the marriage between a patch @AlexeySachkov made and one that I prepared. The solution is still needed to avoid aliases, but it has now been integrated into the visitor, which I indeed think is the better solution. Good thinking!